cfgov/cfgov/tests/test_urls.py

```diff
@@ -133,12 +134,32 @@ def test_handle_error_attribute_error(self, mock_render):
 
     @mock.patch('cfgov.urls.render')
     def test_handle_error(self, mock_render):
-        request = self.factory.get('/test')
+        request = self.factory.get('/Test')
         urls.handle_error(404, request)
         mock_render.assert_called_with(
             request, '404.html', context={'request': request}, status=404
         )
 
+    @mock.patch('cfgov.urls.render')
+    def test_handle_404_error(self, mock_render):
+        request = self.factory.get('/test')
+        urls.handle_404_error(404, request)
+        mock_render.assert_called_with(
+            request, '404.html', context={'request': request}, status=404
+        )
+
+    def test_handle_404_error_case_insensitive_redirect(self):
+        request = self.factory.get('/TEst')
+        response = urls.handle_404_error(404, request)
+        self.assertEqual(response.status_code, 301)
+        self.assertIsInstance(response, HttpResponsePermanentRedirect)
```

Comment:
I think this is unnecessary, given that the status code is 301. A user's browser won't care which Django response object it is. I think it'd be best to restore the `assertRedirects` below instead, to make sure that `/TEst/` redirects to `/test/` — that it redirects from mixed case to lower case.

Reply:
A 301 redirect means that the page has permanently moved to a new location. - This is requested and implemented
A 302 redirect means that the move is only temporary. - This is not requested for case-insensitive URLs

When I was reading: https://docs.djangoproject.com/en/stable/ref/contrib/redirects/
I tried changing `MIDDLEWARE` in `cfgov/settings/base.py` to add:
`django.contrib.redirects.middleware.RedirectFallbackMiddleware`
Currently the following redirect middleware is being used:
`wagtail.contrib.redirects.middleware.RedirectMiddleware`


This seems like something we should pair on.
When I uncomment the self.assertRedirects line I see the following:
```
FAIL: test_handle_404_error_case_insensitive_redirect (cfgov.tests.test_urls.HandleErrorTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "cfgov-refresh/cfgov/cfgov/tests/test_urls.py", line 160, in test_handle_404_error_case_insensitive_redirect
    fetch_redirect_response=True
  File "cfgov-refresh/.tox/unittest-current/lib/python3.6/site-packages/django/test/testcases.py", line 294, in assertRedirects
    % (response.status_code, status_code)
AssertionError: 301 != 302 : Response didn't redirect as expected: Response code was 301 (expected 302)
```
I have tried setting `fetch_redirect_response=False` and `fetch_redirect_response=True`

Reply:
Resolved using
`self.assertEqual(response.get('location'), '/test')`